File: backend/app/api/v1/schemas/ai.py
"""Pydantic schemas for AI chat API."""

from pydantic import BaseModel, Field


class ChatRequest(BaseModel):
    """Request body for POST /api/ai/chat."""

    message: str = Field(..., min_length=1, max_length=32_768)
    conversation_id: str | None = Field(None, description="Optional conversation ID for multi-turn context.")


class ChatResponse(BaseModel):
    """Response body for POST /api/ai/chat."""

    response: str
    conversation_id: str
    trace_id: str | None = None

File: frontend/src/api/ai.ts
import client from "./client";

export interface ChatRequest {
  message: string;
  conversation_id?: string | null;
}

export interface ChatResponse {
  response: string;
  conversation_id: string;
  trace_id?: string | null;
}

export const chat = async (
  message: string,
  conversationId?: string | null,
): Promise<ChatResponse> => {
  const response = await client.post<ChatResponse>("/api/ai/chat", {
    message,
    conversation_id: conversationId ?? undefined,
  });
  return response.data;
};
